### PR TITLE
Migrate plugin-config commands

### PR DESCRIPTION
**Pull request by iowillhoit**
### What does this PR do?
Started with `plugin-template-sf` as a base
Migrated `plugin-config` `v2` commands 

### What issues does this PR fix or reference?
[@W-11572657@](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-11572657)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,67 +1,70 @@
...
   },
   "devDependencies": {
     "@oclif/plugin-command-snapshot": "^3.1.3",
-    "@oclif/test": "^2.1.0",
-    "@salesforce/cli-plugins-testkit": "^1.5.35",
+    "@oclif/test": "^2.1.1",
+    "@salesforce/cli-plugins-testkit": "^2.3.10",
     "@salesforce/dev-config": "^3.1.0",
-    "@salesforce/dev-scripts": "^2",
+    "@salesforce/dev-scripts": "^2.0.4",
```
Comment (mshanemc) [inline package.json:18]: there's a newer dev-scripts that updates the general (non-plugin) linter rules.  Using that now saves some extra work later.

It'll bump some other deps and clean up unused deps, too.

File: src/commands/config/set.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  public static configurationVariablesSection = CONFIG_HELP_SECTION;
+
+  public async run(): Promise<ConfigResponses> {
+    const { flags } = await this.parse(Set);
+    const config: Config = await this.loadConfig(flags.global);
+    let value = '';
+    const configs = await this.parseConfigKeysAndValues();
+    for (const name of Object.keys(configs)) {
+      try {
+        value = configs[name];
```
Comment (mshanemc) [inline src/commands/config/set.ts:38]: avoids `let`, probably safe outside try/catch
```suggestion
    const configs = await this.parseConfigKeysAndValues();
    for (const name of Object.keys(configs)) {
      const value = configs[name] ?? '';
      try {
```

File: src/hooks/init/load_config_meta.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    .map((plugin) => {
+      const configMeta = loadConfigMeta(plugin);
+      if (!configMeta) {
+        log.info(`No config meta found for ${plugin.name}`);
+      }
+
+      return configMeta;
+    })
+    .filter(isObject)
+    .flat();
```
Comment (mshanemc) [inline src/hooks/init/load_config_meta.ts:61]: I *think* the filter is trying to remove `undefined` returned from the `map`, so flatMap should do that in one operation.  That's worth verifying.  

```suggestion
    .flatMap((plugin) => {
      const configMeta = loadConfigMeta(plugin);
      if (!configMeta) {
        log.info(`No config meta found for ${plugin.name}`);
      }

      return configMeta;
    })
```

File: test/.eslintrc.js [hunk 4]
```diff
@@ -10,6 +10,7 @@ module.exports = {
   // Allow describe and it
   env: { mocha: true },
   rules: {
+    '@typescript-eslint/no-misused-promises': 'off',
```
Comment (mshanemc) [inline test/.eslintrc.js:13]: Are these fixable while this move is happening?  Or at least some of them?

**Review by mshanemc — COMMENTED** [review]
going to open a PR for some of this

**Comment by mshanemc** [comment]
some feedback in a PR: https://github.com/salesforcecli/plugin-settings/pull/5

File: src/commands/config/unset.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  public static readonly strict = false;
+
+  public static configurationVariablesSection = CONFIG_HELP_SECTION;
+
+  public static readonly flags = {
+    global: Flags.boolean({
+      char: 'g',
+      summary: messages.getMessage('flags.global.summary'),
+    }),
+    hola: Flags.boolean({
```
[the remark this answers is not in the document]
Comment (iowillhoit) [reply src/commands/config/unset.ts:28]: Whoops. I was QAing another PR with this 🤦‍♂️

**Review by mshanemc — APPROVED** [review]

**Pull request merged**